pkg/assembler/backends/ent/backend/vulnerability.go

```diff
@@ -93,31 +107,45 @@ func getVulnerabilities(ctx context.Context, client *ent.Client, filter model.Vu
 }
 
 func upsertVulnerability(ctx context.Context, client *ent.Tx, spec model.VulnerabilityInputSpec) (*int, error) {
-	// Ingest the type record and ignore if it already exists
+
 	typeId, err := client.VulnerabilityType.Create().
 		SetType(strings.ToLower(spec.Type)).
 		OnConflict(sql.ConflictColumns(vulnerabilitytype.FieldType)).
-		Ignore().
+		DoNothing().
 		ID(ctx)
+
 	if err != nil {
-		return nil, err
+		if err != stdsql.ErrNoRows {
+			return nil, errors.Wrap(err, "upsert vulnerability")
+		}
+		typeId, err = client.VulnerabilityType.Query().
+			Where(vulnerabilitytype.TypeEQ(strings.ToLower(spec.Type))).
+			OnlyID(ctx)
+		if err != nil {
+			return nil, errors.Wrap(err, "get vulnerability type")
+		}
 	}
 
-	vulnInsert := client.VulnerabilityID.Create()
-	vulnID := strings.ToLower(spec.VulnerabilityID)
-
-	vulnInsert.
+	_, err = client.VulnerabilityID.Create().
```

Comment:
I think this should be changed in order to return the ID  of `VulnerabilityID` rather than the `VulnerabilityType` one.
It's basically the same change done for other "nested" resources like Package where previous implementation was returning the `PackageType` ID and now it returns the `PackageVersion` one

Reply:
ok, changed the vulnerability accordingly